Cargo.toml

```diff
@@ -17,6 +17,33 @@ members = [
 [profile.dev]
 panic = "abort"
 
+# Speed up tests by optimizing performance-critical crates
+
+[profile.dev.package.blake2b_simd]
+opt-level = 3
+
+[profile.dev.package.ff]
+opt-level = 3
+
+[profile.dev.package.group]
+opt-level = 3
+
+[profile.dev.package.pasta_curves]
+opt-level = 3
+
+[profile.dev.package.halo2]
+opt-level = 3
+
+[profile.dev.package.bls12_381]
+opt-level = 3
```

Comment:
Question: Is `opt-level = 3` the best compilation time vs test run time tradeoff here?

Here are some tradeoffs I can think of:
* if `opt-level = 1` runs tests almost as fast, we should use it instead, because compilation will be faster
* if the crates are small, the difference might not matter
* if the binary size savings from `opt-level = 3` are large, it might actually be faster
* if this is what `zcashd` does, we might want to do the same thing

Reply:
I was originally testing with `blake2d_simd` and downloading/hashing parameters, and `3` was much faster than `1` (can't remember exactly how much, but more than 2x). I used the same for the CPU-heavy crates since I believe they will act similarly, but it would take a bit of time to actually measure the impact for each one of them.

Comparing these two runs: 

- https://github.com/ZcashFoundation/zebra/actions/runs/1565357849 (this PR)
- https://github.com/ZcashFoundation/zebra/actions/runs/1564710435 (a `main` run at the time)

The build time was actually faster with this PR (could be a CI fluke, but very likely to be _not_ slower at least). The overall CI time was more than 2x faster.

Also, in my machine:

- With those crates in `opt-level = 3`: Build 1m38s, tests: 6m18s
- With those crates in `opt-level = 1`: Build 1m41s, tests: 7m34
- With those crates in `opt-level = 3` and all other dependencies in `opt-level = 1`: Build 2m19s, tests: 4m15s

The gains are not as big, probably because of higher number of cores in my machine. But it does show that it's faster.

I believe [`zcashd` always compiles `librustzcash` with `--release`](https://github.com/zcash/zcash/blob/ae9266a1cd078cd680b9590bd91327fb337d808e/src/Makefile.am#L43), which uses[ `opt-level = 3`](https://doc.rust-lang.org/cargo/reference/profiles.html#release).

As shown above, we can also speed things up by compiling all other dependencies with `opt-level = 1`, but that does make compilation slower so it's not clear how big the savings would be in CI.

Reply:
Thanks, that makes sense!

> As shown above, we can also speed things up by compiling all other dependencies with `opt-level = 1`, but that does make compilation slower so it's not clear how big the savings would be in CI.

I've tried full optimisation before, it significantly increases CI time. (By about 10-20 minutes.)